cmd/embedded-cluster/reset.go

```diff
@@ -116,14 +116,33 @@ func (h *hostInfo) checkQuorumSafety(c *cli.Context) (bool, string, error) {
 	if err != nil {
 		return false, "", fmt.Errorf("unable to unmarshal etcd member list, %w, %s", err, out)
 	}
+	// get a rough picture of the cluster topology
+	workers := []string{}
+	controllers := []string{}
+	nodeList := corev1.NodeList{}
+	err = h.Kclient.List(c.Context, &nodeList)
+	if err != nil {
+		return false, "", fmt.Errorf("unable to create kubernetes client: %w", err)
+	}
+	for _, node := range nodeList.Items {
+		labels := node.GetLabels()
+		if labels["node-role.kubernetes.io/control-plane"] == "true" {
+			controllers = append(controllers, node.Name)
+		} else {
+			workers = append(workers, node.Name)
+		}
+	}
+	if len(workers) > 0 && len(controllers) == 1 {
+		return false, "Cluster has a worker node joined and you are attempting to remove the only controller node.", nil
+	}
 	if len(etcd.Members) < 3 {
 		return true, "", nil
 	}
 	if len(etcd.Members) == 3 {
-		return false, "cluster has 3 control-plane nodes, removing this node will cause etcd to lose quorum", nil
+		return false, "Cluster has 3 control-plane nodes, removing this node will cause etcd to lose quorum.", nil
 	}
 	if len(etcd.Members)%2 != 0 {
-		return false, "cluster would have even number of control-plane nodes after resetting this node, this could cause etcd to become unstable", nil
+		return false, "Cluster would have even number of control-plane nodes after resetting this node, this could cause etcd to become unstable.", nil
```

Comment:
Same comment about controller node role name if possible.

What do you think about saying it would leave the cluster unstable instead of etcd, since we're abstracting away the Kubernetes stuff, and the user likely doesn't know about etcd?
```suggestion
		return false, "This will leave the cluster with an even number of control plane nodes, which could make it unstable.", nil
```